rust/lance-encoding/src/decoder.rs

```diff
@@ -1760,7 +1761,7 @@ impl StructuralBatchDecodeStream {
                 let emitted_batch_size_warning = slf.emitted_batch_size_warning.clone();
                 let task = async move {
                     let next_task = next_task?;
-                    next_task.into_batch(emitted_batch_size_warning)
+                    spawn_cpu(move || next_task.into_batch(emitted_batch_size_warning)).await
```

Comment:
Instead of adding a new `spawn_cpu` can we change the `spawn` here into a `spawn_cpu`?

https://github.com/lancedb/lance/blob/3f74e8564d9b0b890382e78cb3d8ecd28d3ae143/rust/lance-table/src/utils/stream.rs#L383